core/src/main/scala/akka/persistence/r2dbc/journal/R2dbcJournal.scala

```diff
@@ -164,8 +171,26 @@ private[r2dbc] final class R2dbcJournal(config: Config, cfgPath: String) extends
         atomicWrite(batch).map(_ => Nil)(ExecutionContexts.parasitic)
       }
     writesInProgress.put(persistenceId, writeResult)
-    writeResult.onComplete { _ =>
-      context.self ! WriteFinished(persistenceId, writeResult)
+    writeResult.onComplete {
+      case Success(_) if pubSub.isDefined =>
+        val seqNr = messages.head.lowestSequenceNr
+        query
+          .timestampOf(persistenceId, seqNr)
```

Comment:
I believe we can remove this extra query if we use inserts with `returning` syntax. 

For example:

```sql
postgres=# create table foo (
name VARCHAR(255) not null
);
CREATE TABLE

postgres=# INSERT INTO foo(name) VALUES ('bar') returning transaction_timestamp();
     transaction_timestamp
-------------------------------
 2021-12-02 19:58:00.491517+00
(1 row)
```

This syntax is only for postgres, so for other databases (if we ever add support for others) we could then run that extra query. 

For batch inserts, it will return a list of intervals, but we only need one anyway since they will all be the same.

Reply:
Thanks for the suggestion, I have implemented that in https://github.com/akka/akka-persistence-r2dbc/pull/168/commits/a6d4ada76bfb67361ff78f811004a3c0afe93099